src/models/iDocument.ts

```diff
@@ -214,20 +214,26 @@ abstract class iDocument<Type extends DocumentType> {
     }
 
     @action
-    _save() {
+    _save(onBeforeSave: () => Promise<void> = () => Promise.resolve()) {
```

Comment:
new concept - if work needs to be done before saving (as generating the preview-image of excalidraw), a Promise can be passet to `model.save` and it will be called only when the debounced `_save` is called.